airsonic-main/src/main/java/org/airsonic/player/service/MediaScannerService.java

```diff
@@ -240,23 +274,21 @@ private void scanFile(MediaFile file, MusicFolder musicFolder, Instant lastScann
         indexManager.index(file);
 
         if (file.isDirectory()) {
-            for (MediaFile child : mediaFileService.getChildrenOf(file, true, false, false, false)) {
-                scanFile(child, musicFolder, lastScanned, albumCount, genres, isPodcast);
-            }
-            for (MediaFile child : mediaFileService.getChildrenOf(file, false, true, false, false)) {
-                scanFile(child, musicFolder, lastScanned, albumCount, genres, isPodcast);
-            }
+            ForkJoinTask.invokeAll(
+                    mediaFileService.getChildrenOf(file, true, true, false, false)
+                        .parallelStream()
+                        .map(child -> ForkJoinTask.adapt(() -> scanFile(child, musicFolder, lastScanned, albumCount, artists, albums, genres, isPodcast)))
+                        .collect(Collectors.toList()));
```

Comment:
fork into more tasks and submit more tasks to the pool (each child becomes another task)